### DISCUSSION TITLE
How can I configure the API Server?

### DISCUSSION THREAD

**Discussion by zypriafl** [kube-hetzner/terraform-hcloud-kube-hetzner#481] [Q&A] [answered]
How can I configure the API Server as described in https://kubernetes.io/docs/reference/access-authn-authz/authentication/#configuring-the-api-server?

**Comment by captnCC** [comment, marked as answer]
I haven't tested it but it should be possible. To achieve this, the `--oidc-xxx` options need to wrapped with `--kube-apiserver-arg` for k3s. Using the `control_planes_custom_config` variable on the cluster module should allow you to add these config entries. (This probably only works a newly created cluster)

See the docs for more:
- https://docs.k3s.io/reference/server-config#customized-flags-for-kubernetes-processes
- https://docs.k3s.io/installation/configuration#configuration-file

**Comment by zypriafl** [comment]
Thanks for your explanation, we will try that.

**Comment by gigo1980** [comment]
Will be also a solution to change this for an existing cluster?
Or if not with terraform, how could be this done in an other way?

**Reply by mysticaltech** [reply]
@gigo1980 You can try, but I would guess it will not be changed for existing clusters. If it doesn't work, you need to SSH into each control-plane node, go to `/etc/rancher/k3s/config.yaml` and edit it with the params you want, they you need to restart the k3s process with s`ystemctl restart k3s`.

But first, make sure to try via kube.tf options listed above, it could work.
